apps/examples/performance/tls_handshake/tls_handshake_client.c

```diff
@@ -0,0 +1,277 @@
+/****************************************************************************
+ *
+ * Copyright 2021 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <time.h>
+
+#include "mbedtls/config.h"
+#include "mbedtls/net_sockets.h"
+#include "mbedtls/debug.h"
+#include "mbedtls/ssl.h"
+#include "mbedtls/entropy.h"
+#include "mbedtls/ctr_drbg.h"
+#include "mbedtls/error.h"
+#include "mbedtls/certs.h"
+
+#include <string.h>
+
+#define SERVER_PORT "4433"
+static char *SERVER_ADDR = NULL;
+#define GET_REQUEST "GET / HTTP/1.0\r\n\r\n"
+
+#define DEBUG_LEVEL 0
+
+#define mbedtls_printf printf
+
+static void my_debug(void *ctx, int level,
+					 const char *file, int line,
+					 const char *str)
+{
+	((void)ctx);
+	((void)level);
+
+	printf("%s:%04d: %s", file, line, str);
+	fflush((FILE *)ctx);
+}
+
+static unsigned char rootca[] =
+	"-----BEGIN CERTIFICATE-----\r\n"
+	"MIICLTCCAdOgAwIBAgIBATAKBggqhkjOPQQDAjCBjTELMAkGA1UEBhMCS1IxLjAs\r\n"
+	"BgNVBAoMJVNhbXN1bmcgRWxlY3Ryb25pY3MgRGlnaXRhbCBBcHBsaWFuY2UxHDAa\r\n"
+	"BgNVBAsME0RBIFRlc3QgT0NGIFJvb3QgQ0ExMDAuBgNVBAMMJ0RBIFRlc3QgU2Ft\r\n"
+	"c3VuZyBFbGVjdHJvbmljcyBPQ0YgUm9vdCBDQTAgFw0yMTA4MTMwNDIzMzBaGA8y\r\n"
+	"MDY5MTIzMTA0MjMzMFowgY0xCzAJBgNVBAYTAktSMS4wLAYDVQQKDCVTYW1zdW5n\r\n"
+	"IEVsZWN0cm9uaWNzIERpZ2l0YWwgQXBwbGlhbmNlMRwwGgYDVQQLDBNEQSBUZXN0\r\n"
+	"IE9DRiBSb290IENBMTAwLgYDVQQDDCdEQSBUZXN0IFNhbXN1bmcgRWxlY3Ryb25p\r\n"
+	"Y3MgT0NGIFJvb3QgQ0EwWTATBgcqhkjOPQIBBggqhkjOPQMBBwNCAAS2H+PCRt/X\r\n"
+	"7HDgY4inei2uCKsYTq5mcVsaumMHtXqNrs1LF6M0T73hcbjwdFvGhHRPYLBrJRZy\r\n"
+	"UmQwhYTy0KKuoyAwHjAPBgNVHRMBAf8EBTADAQH/MAsGA1UdDwQEAwICxDAKBggq\r\n"
+	"hkjOPQQDAgNIADBFAiBlSErIUCMyKg75TSXQt47WctpwO57cFy398AMl1b+RpAIh\r\n"
+	"AOh+ajEBIKgHNSm6amXOCTBg40J97MBfJflm2DEHLP6v\r\n"
+	"-----END CERTIFICATE-----\r\n";
+
+static int rootca_len = sizeof(rootca);
+
+int tls_handshake_client(char *ipaddr)
+{
+	mbedtls_net_context server_fd;
+	uint32_t flags;
+	unsigned char buf[1024];
+	const char *pers = "ssl_client1";
+
+	mbedtls_entropy_context entropy;
+	mbedtls_ctr_drbg_context ctr_drbg;
+	mbedtls_ssl_context ssl;
+	mbedtls_ssl_config conf;
+	mbedtls_x509_crt cacert;
+	int ret = 1;
+	int len = 0;
+	struct timespec ts;
+	SERVER_ADDR = ipaddr;
+	ts.tv_sec = 1612483200; // 2021-02-05 00:00:00
+	ts.tv_nsec = 0;
+
+	clock_settime(CLOCK_REALTIME, &ts);
```

Comment:
The time is set '2021-02-05 00:00:00' at here, but the certificates are,
```
Validity
Not Before: Aug 13 04:23:30 2021 GMT
Not After : Dec 31 04:23:30 2069 GMT
```

I think the date should be set 'Sep / Oct' for test